src/lib/jsonParseV2/parseEntityObject.cpp

```diff
@@ -52,6 +52,9 @@ std::string parseEntityObject
 {
   std::string type  = jsonParseTypeNames[valueP->GetType()];
 
+  rapidjson::Document  document;
+  document.Parse(ciP->payload);
```

Comment:
Note that the document has been already parsed. The calls sequence is as follows

parseBathUpdate() -> parseEntityVector()-> parseEntityObject()

and parseBathUpdate() is the one that does the parsing.

Thus, to avoid re-parsing here, I'd suggest to get the actionType at parseBatchUpdate() and based on it include a parameters `bool idPatternAllowed`, to be passed to parseEntityObject() in sequence, so you can use it to check if an error has to be returned or not.

Reply:
Fixed in 4c6d654